File: source-base/sep.backend.v1/Data/Entities/BusStop.cs
namespace sep.backend.v1.Data.Entities
{
    public class BusStop : BaseEntity
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public TimeSpan PickUpTime { get; set; }
        public TimeSpan ReturnTime { get; set; }
        public string Address { get; set; }
        public int BusRouteId { get; set; }
        public int Status { get; set; }
        public int Index { get; set; }
        public virtual BusRoute? Route { get; set; }
        public virtual ICollection<BusRouteRegistration>? Registrations { get; set; }
        public virtual ICollection<BusEnrollment>? BusEnrollments { get; set; }
    }
}

File: source-base/source.base.fe/src/types/model/bus-stop.ts
export interface ResponseBusStopData {
  pageNumber: number;
  pageSize: number;
  firstPage: string;
  lastPage: string;
  totalPages: number;
  totalRecords: number;
  nextPage: string | null;
  previousPage: string | null;
  data: BusStop[];
}

export interface RequestBusStopIndex {
  pageNumber: number | null;
  pageSize: number | null;
  status?: number | null;
  keyword?: string | null;
  busRouteId: number;
}

export interface BusStop {
  id: number;
  name: string;
  pickUpTime: string;
  returnTime: string;
  address: string;
  busRouteId: number;
  status: number;
  index: number;
}
export interface CreateBusStop {
  name?: string;
  pickUpTime?: string;
  returnTime?: string;
  address?: string;
  busRouteId?: number;
  status?: number;
}

export interface ErrorResponseCreateBusStop {
    Name?: string[];
    PickUpTime: string[];
    ReturnTime: string[];
    Address?: string[];
    BusRouteId?: string[];
    Status?: string[];
}
